assets/src/stories-editor/helpers/index.js

```diff
@@ -1715,3 +1734,121 @@ export const processMedia = ( media ) => {
+
+		const widthFactor = STORY_PAGE_INNER_WIDTH > width ? STORY_PAGE_INNER_WIDTH / width : 1;
+		const heightFactor = STORY_PAGE_INNER_HEIGHT > height ? STORY_PAGE_INNER_HEIGHT / height : 1;
+
+		return Math.max( widthFactor, heightFactor ) * scalingFactor;
+	}
+
+	return 1;
+};
+
+/**
+ * Plays the block's animation in the editor.
+ *
+ * @param {Object} block Block object.
+ * @param {string} animationType Animation type.
+ * @param {number} animationDuration Animation duration.
+ * @param {number} animationDelay Animation delay.
+ * @param {function} callback Callback for when animation has stopped.
+ */
+export const playAnimation = ( block, animationType, animationDuration, animationDelay, callback ) => {
+	const blockElement = getBlockInnerElementForAnimation( block );
+	const parentBlock = getBlockRootClientId( block.clientId );
+	const parentBlockElement = document.querySelector( `[data-block="${ parentBlock }"]` );
+
+	if ( ! blockElement || ! parentBlock || ! parentBlockElement ) {
+		if ( callback ) {
+			callback();
+		}
+
+		return;
+	}
+
+	const DEFAULT_ANIMATION_DURATION = ANIMATION_DURATION_DEFAULTS[ animationType ] || 0;
+	const animationName = `story-animation-${ animationType }`;
+
+	blockElement.classList.remove( animationName );
+
+	blockElement.style.setProperty( '--animation-duration', `${ animationDuration || DEFAULT_ANIMATION_DURATION }ms` );
+	blockElement.style.setProperty( '--animation-delay', `${ animationDelay }ms` );
+
+	const { left: parentBlockOffsetLeft, top: parentBlockOffsetTop } = parentBlockElement.getBoundingClientRect();
+	const { top, left, width, height } = blockElement.getBoundingClientRect();
+	const scalingFactor = calculateTargetScalingFactor( width, height );
+
+	// We calculate with the block's actual dimensions relative to the page it's on.
+	const actualTop = top - parentBlockOffsetTop;
+	const actualLeft = left - parentBlockOffsetLeft;
+
+	let offsetX;
+	let offsetY;
+
+	// @todo Verify this.
+	switch ( animationType ) {
+		case 'fly-in-left':
+		case 'rotate-in-left':
+		case 'whoosh-in-left':
+			offsetX = -( actualLeft + width );
+			break;
+		case 'fly-in-right':
+		case 'rotate-in-right':
+		case 'whoosh-in-right':
+			offsetX = STORY_PAGE_INNER_WIDTH + actualLeft + width;
+			break;
+		case 'fly-in-top':
+			offsetY = -( actualTop + height );
+			break;
+		case 'fly-in-bottom':
+			// const offsetY = dimensions.pageHeight - dimensions.targetY;
+			offsetY = STORY_PAGE_INNER_HEIGHT + actualTop + height;
+			break;
+		case 'drop':
+			offsetY = Math.max( 160, ( actualTop + height ) );
+			break;
+		case 'pan-left':
+		case 'pan-right':
+			offsetX = STORY_PAGE_INNER_WIDTH - ( width * scalingFactor );
+			offsetY = ( STORY_PAGE_INNER_HEIGHT - ( height * scalingFactor ) ) / 2;
+			blockElement.style.setProperty( '--animation-scale-start', scalingFactor );
+			blockElement.style.setProperty( '--animation-scale-end', scalingFactor );
+			break;
+		case 'pan-down':
+		case 'pan-up':
```

Comment:
Since images with a `pan-*` animation are always scaled on the frontend (even after animation has finished), I wonder whether we should do the same in the editor as well for a better WYSIWYG experience.

Reply:
Yes, would be great! Are there any downsides?

Reply:
It might be a bit unexpected for the user perhaps. I'll do a quick check to see how feasible it is.

Reply:
We couldn't just use `scale()` in CSS, as that would scale _everything_, including things like resize handles. Also, the image must not overflow the page in that case.

We'd need to actually increase the width and height in the editor by the determined scaling factor, but only in the editor and when not playing the animation preview.

Since this is a bit out of scope for this issue/PR here, I'll open a new one to explore this.

Reply:
Issue opened: #3118.